crates/ruff/src/rules/ruff/rules/pairwise_over_zipped.rs

```diff
@@ -0,0 +1,114 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::source_code::Stylist;
+use rustpython_parser::ast::{Expr, ExprKind};
+
+use crate::checkers::ast::Checker;
+use crate::settings::types::PythonVersion;
+use crate::Range;
+use ruff_python_ast::helpers::unparse_constant;
+
+#[violation]
+pub struct PairwiseOverZipped {}
+
+impl Violation for PairwiseOverZipped {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Prefer `itertools.pairwise()` to `zip()` if trying to fetch an iterable's successive overlapping pairs.")
+    }
+}
+
+#[derive(Debug)]
+struct SliceInfo {
+    arg_name: String,
+    slice_start: i32,
+    slice_end: i32,
```

Comment:
I'm not as familiar with python but should these be `i64` to support a broader range of valid int indices?

Reply:
Done